src/main/java/net/authorize/sample/TransactionReporting/GetAccountUpdaterJobDetails.java

```diff
@@ -0,0 +1,154 @@
+import net.authorize.api.contract.v1.MessageTypeEnum;
+import net.authorize.api.contract.v1.Paging;
+import net.authorize.api.controller.GetAUJobDetailsController;
+import net.authorize.api.controller.base.ApiOperationBase;
+
+public class GetAccountUpdaterJobDetails {
+
+	public static ANetApiResponse run(String apiLoginId, String transactionKey) {
+
+		// Set the request to operate in either the sandbox or production environment
+		ApiOperationBase.setEnvironment(Environment.SANDBOX);
+
+		// Create object with merchant authentication details
+		MerchantAuthenticationType merchantAuthenticationType = new MerchantAuthenticationType();
+		merchantAuthenticationType.setName(apiLoginId);
+		merchantAuthenticationType.setTransactionKey(transactionKey);
+
+		String month = "2018-08";
+		String reFid = "123456";
+		Paging paging = new Paging();
+		paging.setLimit(100);
+		paging.setOffset(2);
+		
+
+		// Create the API request and set the parameters for this specific request
+		GetAUJobDetailsRequest apiRequest = new GetAUJobDetailsRequest();
+		apiRequest.setMerchantAuthentication(merchantAuthenticationType);
+		apiRequest.setPaging(paging);
+		apiRequest.setRefId(reFid);
+		apiRequest.setMonth(month);
+		apiRequest.setModifiedTypeFilter(AUJobTypeEnum.ALL);
+
+		// Call the controller
+		GetAUJobDetailsController controller = new GetAUJobDetailsController(apiRequest);
+		controller.execute();
+
+		GetAUJobDetailsResponse response = new GetAUJobDetailsResponse();
+		response = controller.getApiResponse();
+
+		// If API Response is OK, go ahead and check the transaction response
+		if (response != null && response.getMessages().getResultCode() == MessageTypeEnum.OK) {
+
+			System.out.println("SUCCESS: Get Account Updater job details for Month: " + month);
+
+			if (response.getAuDetails() == null) {
+				System.out.println("No GetAccountUpdaterjobdetails for Month.");
+				return response;
+			}
+
+			ArrayList<AuUpdateType> updateTypeList = new ArrayList<AuUpdateType>();
+			ArrayList<AuDeleteType> deleteTypeList = new ArrayList<AuDeleteType>();
+
+			for (AuDetailsType details : response.getAuDetails().getAuUpdateOrAuDelete()) {
+
+				System.out.println("---Customer Profile Details Start---");
+				System.out.println("Profile ID:" + details.getCustomerProfileID());
+				System.out.println("Payment Profile ID:" + details.getCustomerPaymentProfileID());
+				System.out.println("First Name:" + details.getFirstName());
+				System.out.println("Last Name:" + details.getLastName());
+				System.out.println("Update Time UTC:" + details.getUpdateTimeUTC());
+				System.out.println("Reason Code:" + details.getAuReasonCode());
+				System.out.println("Reason Description:" + details.getReasonDescription());
+
+				if (details.getClass().getTypeName().toString().contains("AuUpdateType")) {
+
+					updateTypeList.add((AuUpdateType) details);
+
+				} else if (details.getClass().getTypeName().toString().contains("AuDeleteType")) {
```

Comment:
These samples codes need to be compatible with Java 7. The method getTypeName() is available only from Java 8. This needs to be re-worked.

Reply:
Resolved the comment.